### PR TITLE
feat(accordion): remove shim mapping and fallbacks

### PR DESCRIPTION
**Pull request by mivaylo**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: CDE-1257 CDE-1262

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

Since accordion and stepper are relatively big components we want to focus solely on the PR for removing the shim, and there is a refactoring task for leftovers and optimizing the codebase of these two components:

https://jira.eng.vmware.com/browse/CDE-1476

### CODE REVIEW

File: projects/angular/src/accordion/_variables.accordion.scss [hunk 1]
```diff
@@ -4,34 +4,40 @@
...
-$clr-accordion-padding: variables.$clr_baselineRem_18px !default;
-$clr-accordion-border-radius: variables.$clr-global-borderradius !default;
-$clr-accordion-border-left-width: variables.$clr_baselineRem_0_25 !default;
-$clr-accordion-border-left-color-complete: $clr-accordion-complete-color !default;
-$clr-accordion-border-left-color-error: $clr-accordion-error-color !default;
+$clr-accordion-border-color: var(--clr-accordion-border-color) !default;
+$clr-accordion-border-radius: var(--clr-accordion-border-radius) !default;
+$clr-accordion-border-left-color-complete: var(--clr-accordion-border-left-color-complete) !default;
+$clr-accordion-border-left-color-error: var(--clr-accordion-border-left-color-error) !default;
 $clr-accordion-content-color: variables.$clr-p1-color !default;
```
Comment (dtsanevmw) [inline projects/angular/src/accordion/_variables.accordion.scss:39]: Is this needed? Use equivalent custom property with a core token value or remove.

File: projects/angular/src/accordion/_variables.accordion.scss [hunk 2]
```diff
@@ -4,34 +4,40 @@
...
+$clr-accordion-header-left-current-indicator: var(--clr-accordion-header-left-current-indicator) !default;
+$clr-accordion-header-gap: var(--clr-accordion-header-gap) !default;
+$clr-accordion-content-color: var(--clr-accordion-content-color) !default;
+$clr-accordion-content-font-size: var(--clr-accordion-content-font-size) !default;
+$clr-accordion-content-padding: tokens.$cds-global-space-5 tokens.$cds-global-space-6 tokens.$cds-global-space-5
+  tokens.$cds-global-space-11 !default;
+
 $clr-accordion-responsive-breakpoint: map.get(variables.$clr-grid-breakpoints, sm) !default;
 $clr-accordion-error-color: variables.$clr-global-error-color !default;
 $clr-accordion-complete-color: variables.$clr-global-success-color !default;
```
Comment (dtsanevmw) [inline projects/angular/src/accordion/_variables.accordion.scss:34]: Use respective custom properties and core tokens.

File: projects/angular/src/accordion/_properties.accordion.scss [hunk 3]
```diff
@@ -4,80 +4,49 @@
...
 
-      --clr-accordion-content-color: #{accordion-variables.$clr-accordion-content-color};
-      --clr-accordion-content-font-size: #{accordion-variables.$clr-accordion-content-font-size};
-      --clr-accordion-description-margin-left: #{accordion-variables.$clr-accordion-description-margin-left};
-      --clr-accordion-step-title-min-width: #{accordion-variables.$clr-accordion-step-title-min-width};
-      --clr-accordion-step-title-font-weight: #{accordion-variables.$clr-accordion-step-title-font-weight};
-      --clr-accordion-step-status-margin-left: #{accordion-variables.$clr-accordion-step-status-margin-left};
-      --clr-accordion-step-title-margin-left: #{accordion-variables.$clr-accordion-step-title-margin-left};
+      --clr-accordion-step-status-margin-left: #{tokens.$cds-global-space-7};
+      --clr-accordion-step-title-margin-left: #{tokens.$cds-global-space-9};
```
Comment (dtsanevmw) [inline projects/angular/src/accordion/_properties.accordion.scss:49]: We can just use the core values inside _accordion. Exposed properties are not needed for these

File: projects/angular/src/accordion/_properties.accordion.scss [hunk 4]
```diff
@@ -4,80 +4,49 @@
...
-        var(--clr-accordion-border-left-width)
-        0
-        0
-        var(--clr-accordion-border-left-color-complete);
-      --clr-accordion-header-left-error-indicator: inset
-        var(--clr-accordion-border-left-width)
-        0
-        0
-        var(--clr-accordion-border-left-color-error);
+      --clr-accordion-header-gap: #{tokens.$cds-global-space-5};
```
Comment (dtsanevmw) [inline projects/angular/src/accordion/_properties.accordion.scss:44]: Use `tokens.$cds-global-space-5` directly in the file.

File: projects/angular/src/accordion/_properties.accordion.scss [hunk 5]
```diff
@@ -4,80 +4,49 @@
...
 @include mixins.exports('accordion.properties') {
-  @if variables.$clr-use-custom-properties == true {
-    :root {
-      --clr-accordion-text-color: var(--clr-global-font-color);
-      --clr-accordion-text-color-hover: var(--clr-global-font-color);
-      --clr-accordion-text-color-active: var(--clr-global-font-color);
-      --clr-accordion-text-color-selected: var(--clr-global-font-color);
-      --clr-accordion-panel-disabled-color: var(--clr-color-neutral-500);
+  :root {
+    [cds-theme] {
```
Comment (dtsanevmw) [inline projects/angular/src/accordion/_properties.accordion.scss:13]:
```suggestion
  :root   [cds-theme] {
```

File: projects/angular/src/accordion/_variables.accordion.scss [hunk 6]
```diff
@@ -4,34 +4,40 @@
...
+
+$clr-accordion-header-button-padding: var(--clr-accordion-header-button-padding) !default;
+$clr-accordion-error-color: var(--clr-accordion-error-color) !default;
+$clr-accordion-complete-color: var(--clr-accordion-complete-color) !default;
+$clr-accordion-header-left-indicator: var(--clr-accordion-header-left-indicator) !default;
+$clr-accordion-header-left-current-indicator: var(--clr-accordion-header-left-current-indicator) !default;
+$clr-accordion-header-gap: var(--clr-accordion-header-gap) !default;
+$clr-accordion-content-color: var(--clr-accordion-content-color) !default;
+$clr-accordion-content-font-size: var(--clr-accordion-content-font-size) !default;
+$clr-accordion-content-padding: tokens.$cds-global-space-5 tokens.$cds-global-space-6 tokens.$cds-global-space-5
```
Comment (dtsanevmw) [inline projects/angular/src/accordion/_variables.accordion.scss:29]: Use values directly in the file. Variable not needed.

File: projects/angular/src/accordion/_accordion.clarity.scss [hunk 7]
```diff
@@ -7,160 +7,70 @@
...
-    border-bottom: 0;
+    border: tokens.$cds-global-space-1 solid accordion-variables.$clr-accordion-border-color;
+    border-bottom-width: 0;
     width: 100%;
-    @include mixins.css-var(
-      font-size,
-      clr-accordion-header-font-size,
-      variables.$clr_baselineRem_0_65,
-      variables.$clr-use-custom-properties
-    );
```
Comment (dtsanevmw) [inline projects/angular/src/accordion/_accordion.clarity.scss:60]: I see that property was exposed before TM so you will need to use it as override in the mixin or deprecate it.

File: projects/angular/src/accordion/_accordion.clarity.scss [hunk 8]
```diff
@@ -170,12 +80,12 @@
...
     width: 100%;
     border: 0;
-    padding: accordion-variables.$clr-accordion-padding;
+    padding: tokens.$cds-global-space-8;
     background: transparent;
     text-align: left;
     cursor: pointer;
     color: inherit;
-    @include mixins.css-var(gap, clr-accordion-header-gap, 0, variables.$clr-use-custom-properties);
+    gap: accordion-variables.$clr-accordion-header-gap;
```
Comment (dtsanevmw) [inline projects/angular/src/accordion/_accordion.clarity.scss:88]: Replace with core token,

File: projects/angular/src/accordion/_accordion.clarity.scss [hunk 9]
```diff
@@ -284,15 +145,15 @@
...
 
     @media (min-width: accordion-variables.$clr-accordion-responsive-breakpoint) {
-      padding: variables.$clr_baselineRem_0_75 variables.$clr_baselineRem_2_25;
+      padding: tokens.$cds-global-space-8 mixins.baselinePx(54);
     }
   }
 
   .clr-accordion-inner-content {
-    padding: accordion-variables.$clr-accordion-padding;
+    padding: mixins.baselinePx(18);
```
Comment (dtsanevmw) [inline projects/angular/src/accordion/_accordion.clarity.scss:153]: Should be cds-global-space-9 as specified in figma.

File: projects/angular/src/accordion/_accordion.clarity.scss [hunk 10]
```diff
@@ -460,105 +251,38 @@
...
-        clr-accordion-status-icon-width,
-        variables.$clr_baselineRem_0_8,
-        variables.$clr-use-custom-properties
-      );
+      width: auto;
 
       cds-icon,
       clr-icon {
-        @include mixins.equilateral(variables.$clr_baselineRem_0_583);
+        @include mixins.equilateral(mixins.baselinePx(14));
```
Comment (dtsanevmw) [inline projects/angular/src/accordion/_accordion.clarity.scss:285]: Looks like cds-global-space-7 is used for that.

File: projects/angular/src/accordion/_accordion.clarity.scss [hunk 11]
```diff
@@ -460,105 +251,38 @@
...
-    .clr-accordion-title {
-      @include mixins.css-var(font-weight, clr-accordion-title-font-weight, 500, variables.$clr-use-custom-properties);
-      @include mixins.css-var(
-        font-size,
-        clr-accordion-title-font-size,
-        variables.$clr_baselineRem_0_5416,
-        variables.$clr-use-custom-properties
-      );
-    }
-
```
Comment (dtsanevmw) [inline projects/angular/src/accordion/_accordion.clarity.scss:504]: Why removing the individual styling and apply the typography mixin to the whole header?

File: projects/angular/src/accordion/_accordion.clarity.scss [hunk 12]
```diff
@@ -581,22 +306,55 @@
...
+    }
+
+    .clr-accordion-description {
       display: inline-block;
+      max-width: mixins.baselinePx(650);
+
+      @media (min-width: accordion-variables.$clr-accordion-responsive-breakpoint) {
+        margin-left: tokens.$cds-global-space-12;
+      }
+    }
```
Comment (dtsanevmw) [inline projects/angular/src/accordion/_accordion.clarity.scss:326]: In figma it seems that the description is all the way in the right? Am I missing something?

**Review by dtsanevmw — CHANGES_REQUESTED** [review]
Check comments

File: projects/angular/src/accordion/STYLES.md [hunk 13]
```diff
@@ -2,46 +2,34 @@
 
 ## CSS Custom Properties
 
-| CSS Custom Property                              | Description                                                          |
-| ------------------------------------------------ | -------------------------------------------------------------------- |
-| --clr-accordion-text-color                       | Text color of the accordion component.                               |
-| --clr-accordion-text-color-hover                 | Text color of the accordion header when hovered.                     |
-| --clr-accordion-text-color-active                | Text color of the accordion header when active.                      |
-| --clr-accordion-text-color-selected              | Text color of the accordion header when selected.                    |
```
Comment (valentin-mladenov) [inline projects/angular/src/accordion/STYLES.md:10]: `--clr-accordion-text-color-selected` property should not be removed.

File: projects/angular/src/accordion/_properties.accordion.scss [hunk 14]
```diff
@@ -4,80 +4,49 @@
...
 
-      --clr-accordion-title-font-weight: 500;
-      --clr-accordion-title-font-size: #{variables.$clr_baselineRem_0_5416};
+      --clr-accordion-error-color: #{tokens.$cds-alias-status-danger};
+      --clr-accordion-complete-color: #{tokens.$cds-alias-status-success};
 
-      --clr-accordion-error-color: var(--clr-global-error-color);
-      --clr-accordion-complete-color: var(--clr-global-success-color);
+      --clr-accordion-border-color: #{tokens.$cds-alias-object-border-color};
+      --clr-accordion-border-radius: #{tokens.$cds-global-space-3};
```
Comment (valentin-mladenov) [inline projects/angular/src/accordion/_properties.accordion.scss:35]:
```suggestion
      --clr-accordion-border-radius: #{tokens.$cds-alias-object-border-radius-100};
```

File: projects/angular/src/accordion/_accordion.clarity.scss (hunk 7 above)
Comment (mivaylo) [reply projects/angular/src/accordion/_accordion.clarity.scss:60]: Added "Deprecated" in STYLES.md and "@Deprecated" in the variables file

File: projects/angular/src/accordion/_variables.accordion.scss (hunk 2 above)
Comment (dtsanevmw) [reply projects/angular/src/accordion/_variables.accordion.scss]: Duplicated now

**Review by dtsanevmw — APPROVED** [review]

File: projects/angular/src/accordion/_variables.accordion.scss (hunk 1 above)
Comment (mivaylo) [reply projects/angular/src/accordion/_variables.accordion.scss]: fixed, now it's used in the styles

File: projects/angular/src/accordion/_variables.accordion.scss (hunk 2 above)
Comment (mivaylo) [reply projects/angular/src/accordion/_variables.accordion.scss]: removed

**Pull request merged**